doc/guides/optimizations/tools.rst

```diff
@@ -0,0 +1,209 @@
+    arch                                                                                    1194     5.74%
+    arm                                                                                   1194     5.74%
+        core                                                                                1194     5.74%
+        aarch32                                                                           1194     5.74%
+            cortex_m                                                                         852     4.09%
+            fault.c                                                                        400     1.92%
+                bus_fault.isra.0                                                              60     0.29%
+                mem_manage_fault.isra.0                                                       56     0.27%
+                usage_fault.isra.0                                                            36     0.17%
+                z_arm_fault                                                                  232     1.11%
+                z_arm_fault_init                                                              16     0.08%
+            irq_init.c                                                                      24     0.12%
+                z_arm_interrupt_init                                                          24     0.12%
+            mpu                                                                            352     1.69%
+                arm_core_mpu.c                                                                56     0.27%
+                z_arm_configure_static_mpu_regions                                          56     0.27%
+                arm_mpu.c                                                                    296     1.42%
+                __init_sys_init_arm_mpu_init0                                                8     0.04%
+                arm_core_mpu_configure_static_mpu_regions                                   20     0.10%
+                arm_core_mpu_disable                                                        16     0.08%
+                arm_core_mpu_enable                                                         20     0.10%
+                arm_mpu_init                                                                92     0.44%
+                mpu_configure_regions                                                      140     0.67%
+            thread_abort.c                                                                  76     0.37%
+                z_impl_k_thread_abort
+                76     0.37%
+    ...
+    ...
+
+
+Data Structures
+****************
+
+
+Build Target: pahole
+=====================
+
+Poke-a-hole (pahole) is an object-file analysis tool to find the size of
+the data structures, and the holes caused due to aligning the data
+elements to the word-size of the CPU by the compiler.
```

Comment:
We should probably mention that `pahole` must be installed prior to using this target - and mention where to obtain it (e.g. `https://git.kernel.org/pub/scm/devel/pahole/pahole.git` and perhaps `sudo apt-get install dwarves`).